### PR TITLE
adds external ingress controller

### PR DESCRIPTION
**Pull request by chrisstafford**
Depends on https://github.com/OCP-on-NERC/nerc-ocp-config/pull/47

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202594685849644

### CODE REVIEW

File: cluster-scope/overlays/nerc-ocp-prod/ingresscontrollers/external-apps-ingress-controller.yaml
Comment (naved001) [inline cluster-scope/overlays/nerc-ocp-prod/ingresscontrollers/external-apps-ingress-controller.yaml:9]: where's this certificate created? I don't see it in #47

Comment (chrisstafford) [reply cluster-scope/overlays/nerc-ocp-prod/ingresscontrollers/external-apps-ingress-controller.yaml:9]: This cert does not yet exist, once I have it I'll update this PR

**Comment by joachimweyl** [comment]
@chrisstafford  The Vault is unsealed. What are the next steps to getting this merged?

**Comment by chrisstafford** [comment]
> @chrisstafford The Vault is unsealed. What are the next steps to getting this merged?

The vault was unsealed, but the Infra cluster hosted vault which is down at the moment. I can't create the secret until we get the Infra cluster back up.

File: cluster-scope/overlays/nerc-ocp-prod/ingresscontrollers/external-apps-ingress-controller.yaml
Comment (chrisstafford) [reply cluster-scope/overlays/nerc-ocp-prod/ingresscontrollers/external-apps-ingress-controller.yaml:9]: I've added the secret into https://github.com/OCP-on-NERC/nerc-ocp-config/pull/47

**Comment by joachimweyl** [comment]
@chrisstafford I believe the Infra Cluster hosted Vault issue is resolved. What are the next steps for getting this merged?

**Comment by naved001** [comment]
I will approve this once it's rebased after #47 is merged because that adds the certificate that this references.

**Review by larsks — APPROVED** [review]
This looks like it will create an ingress controller. I'm a little unclear on how we will expose the public network to this controller, but perhaps that's for a different PR.

I'm approving it now, since it looks sane and I'm trying to clear my queue, but as @naved001  said it shouldn't be merged until #47 has merged.

**Review by computate — APPROVED** [review]

**Pull request merged**
